raiden/network/proxies/service_registry.py

```diff
@@ -48,7 +48,7 @@ def __init__(
 
     def ever_made_deposits(
         self, block_identifier: BlockSpecification, index: int
-    ) -> Optional[AddressHex]:
+    ) -> Optional[Address]:
```

Comment:
Ah but this won't work alone like this. You also need to transform the hexadecimal address returned by the call to bytes from below. You need to use `to_canonical_address()`

Reply:
I added `assert isinstance(ret, T_Address)` and trying locally to confirm failure.

Reply:
Nice
```
raiden/network/pathfinding.py:192: error: Argument "service_hex_address" to "get_service_url" of "ServiceRegistry" has incompatible type "Address"; expected "AddressHex"
```

Reply:
Also
```
raiden/network/proxies/service_registry.py:65: error: Incompatible return value type (got "bytes", expected "Optional[Address]")
```

Reply:
Done the necessary changes.